core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CloudflowBasePlugin.scala

```diff
@@ -58,6 +59,58 @@ abstract class CloudflowBasePlugin extends AutoPlugin {
   override def requires = CommonSettingsAndTasksPlugin && StreamletScannerPlugin &&
     JavaAppPackaging && sbtdocker.DockerPlugin
 
+  override def projectSettings = Seq(
+    libraryDependencies ++= Vector(
+      "com.lightbend.cloudflow" % "cloudflow-runner"         % BuildInfo.version
+    ),
+
+    cloudflowDockerImageName := Def.task {
+      Some(DockerImageName((ThisProject / name).value.toLowerCase, (ThisProject / cloudflowBuildNumber).value.buildNumber))
+    }.value,
+
+    streamletDescriptorsInProject := Def.taskDyn {
+      val detectedStreamlets = cloudflowStreamletDescriptors.value
+      buildStreamletDescriptors(detectedStreamlets)
+    }.value,
+
+    buildOptions in docker := BuildOptions(
+      cache = true,
+      removeIntermediateContainers = BuildOptions.Remove.OnSuccess,
+      pullBaseImage = BuildOptions.Pull.IfMissing
+    ),
+
+    imageNames in docker := {
+      val registry = cloudflowDockerRegistry.value
```

Comment:
are these val's used anywhere else?

Reply:
These need to be set for `sbt-docker` plugin for building the docker image

Reply:
So they are an override?

Reply:
They are local vals and so I think @olofwalker is right and e could just inline them since they each seem to be used only once.

Reply:
They cannot be inlined since you cannot have an sbt `TaskKey` evaluation within an anonymous function. Hence I have to evaluate it outside the `.map { ..` ..
<img width="926" alt="Screen Shot 2020-02-12 at 2 36 55 PM" src="https://user-images.githubusercontent.com/107231/74319656-4cb38f00-4da5-11ea-9a1d-e7682790f78f.png">

Reply:
@debasishg Thanks for that clarification, as usual my sbt-fu is lacking